packages/api/src/routers/api/sessions.ts

```diff
@@ -1,112 +1,150 @@
+    try {
+      const { sessionId } = req.params;
+      const { endTime, sourceId, limit, offset, startTime } = req.query;
 
-    const team = await getTeam(teamId);
-    if (team == null) {
-      return res.sendStatus(403);
-    }
+      const { teamId } = getNonNullUserWithTeam(req);
 
-    res.json(
-      await clickhouse.getSessions({
-        endTime: endTimeNum,
-        limit: 500, // fixed limit for now
-        offset: 0, // fixed offset for now
-        q: q as string,
-        startTime: startTimeNum,
-        tableVersion: team.logStreamTableVersion,
-        teamId: teamId.toString(),
-      }),
-    );
-  } catch (e) {
-    const span = opentelemetry.trace.getActiveSpan();
-    span?.recordException(e as Error);
-    span?.setStatus({ code: SpanStatusCode.ERROR });
-
-    next(e);
-  }
-});
-
-router.get('/:sessionId/rrweb', async (req, res, next) => {
-  try {
-    const teamId = req.user?.team;
-    const { sessionId } = req.params;
-    const { endTime, limit, offset, startTime } = req.query;
-    if (teamId == null) {
-      return res.sendStatus(403);
-    }
-    const startTimeNum = parseInt(startTime as string);
-    const endTimeNum = parseInt(endTime as string);
-    const limitNum = parseInt(limit as string);
-    const offsetNum = parseInt(offset as string);
-    if (
-      !isNumber(startTimeNum) ||
-      !isNumber(endTimeNum) ||
-      !isNumber(limitNum) ||
-      !isNumber(offsetNum)
-    ) {
-      return res.sendStatus(400);
-    }
+      const source = await Source.findById(sourceId);
+
+      if (!source) {
+        res.status(404).send('Source not found');
+        return;
+      }
+
+      const connection = await getConnectionById(
+        teamId.toString(),
+        source.connection.toString(),
+        true,
+      );
+
+      if (!connection) {
+        res.status(404).send('Connection not found');
+        return;
+      }
 
-    const MAX_LIMIT = 1e6;
-
-    res.setHeader('Cache-Control', 'no-cache');
-    res.setHeader('Content-Type', 'text/event-stream');
-    res.setHeader('Connection', 'keep-alive');
-    res.flushHeaders(); // flush the headers to establish SSE with client
-
-    const stream = await clickhouse.getRrwebEvents({
-      sessionId: sessionId as string,
-      startTime: startTimeNum,
-      endTime: endTimeNum,
-      limit: Math.min(MAX_LIMIT, limitNum),
-      offset: offsetNum,
-    });
-
-    stream.on('data', (rows: Row[]) => {
-      res.write(`${rows.map(row => `data: ${row.text}`).join('\n')}\n\n`);
-      res.flush();
-    });
-    stream.on('end', () => {
-      logger.info('Stream ended');
-
-      res.write('event: end\ndata:\n\n');
+      const MAX_LIMIT = 1e6;
+
+      res.setHeader('Cache-Control', 'no-cache');
+      res.setHeader('Content-Type', 'text/event-stream');
+      res.setHeader('Connection', 'keep-alive');
+      res.flushHeaders(); // flush the headers to establish SSE with client
+
+      const clickhouseClient = new ClickhouseClient({
```

Comment:
only updated the query part to use new `ClickhouseClient` from common-utils which takes query rendered by v2 renderConfig